### PR TITLE
zero NAs in pillar2 data we're fitting to and trim age-specific pilla…

### PR DESCRIPTION
**Pull request by pabloperguz**
In this PR:

1. NAs in pillar 2 data we're actually fitting to are zeroed
2. Add age-specific pillar 2 columns to trimming 
3. Move age-bands one level up from function `spim_lancelot_data_rtm` to `spim_data_single`

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -96,10 +101,14 @@ spim_data_single <- function(date, region, model_type, rtm, serology,
...
   ## are too likely to be back-filled to be reliable
+  ## TODO: this works only for Lancelot - carehomes will be soon removed
   i <- seq(to = nrow(data), length.out = trim_pillar2)
   cols_pillar2 <- c("pillar2_tot", "pillar2_pos", "pillar2_cases",
                     "pillar2_over25_tot", "pillar2_over25_pos",
-                    "pillar2_over25_cases")
+                    "pillar2_over25_cases",
+                    paste0("pillar2_", pillar2_age_bands, "_tot"),
+                    paste0("pillar2_", pillar2_age_bands, "_pos"),
+                    paste0("pillar2_", pillar2_age_bands, "_cases"))
```
Comment (edknock) [inline R/data.R:111]: is there any reason we can't do all of this with a `grep`?

Comment (pabloperguz) [reply R/data.R]: none... good shout!

File: R/data.R [hunk 2]
```diff
@@ -613,6 +615,12 @@ spim_lancelot_data_rtm <- function(date, region, model_type, data, full_data,
 
   # ignore pillar 2 testing before 2020-06-18
   data[which(data$date < "2020-06-18"), cols_pillar2] <- NA_integer_
+  # after that date turn NAs to zeroes for columns where data is available
+  for (i in cols_pillar2) {
+    if (!all(is.na(data[, i]))) {
+      data[which(data$date >= "2020-06-18"), i] <- 0
+    }
+  }
```
Comment (edknock) [inline R/data.R:623]: This is setting everything to 0 (and not just the NAs)

Comment (edknock) [reply R/data.R:623]: I suggest it be `data[which(is.na(data[, i])), i] <- 0` and then move this above the "ignore pillar 2 testing before 2020-06-18" bit

File: R/data.R [hunk 3]
```diff
@@ -57,14 +57,19 @@ spim_data <- function(date, region, model_type, rtm, serology,
 spim_data_single <- function(date, region, model_type, rtm, serology,
                              trim_deaths, trim_pillar2, full_data,
                              fit_to_variants, sircovid_model) {
+
+  pillar2_over25_age_bands <- c("25_49", "50_64", "65_79", "80_plus")
+  pillar2_age_bands <- c("under15", "15_24", pillar2_over25_age_bands)
```
Comment (edknock) [inline R/data.R:62]: Perhaps now these are only needed in `spim_lancelot_data_rtm`, this could be moved back into there?

Comment (pabloperguz) [reply R/data.R]: Thanks, done

File: R/data.R (hunk 2 above)
Comment (pabloperguz) [reply R/data.R:622]: crap, sorry... mental fart... done

**Review by edknock — APPROVED** [review]

**Review by edknock — APPROVED** [review]

**Pull request merged**